presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -6055,4 +6055,17 @@ public void testSourceLocationInPlan()
         assertTrue(plan.contains("m := max (1:34)"));
         assertTrue(plan.contains("max := max(expr) RANGE UNBOUNDED_PRECEDING CURRENT_ROW (1:34)"));
     }
+
+    @Test
+    public void testCSECompilation()
+    {
+        String sql = "SELECT " +
+                "FILTER( MAP_VALUES(map1), x -> x >= ELEMENT_AT( MAP(ARRAY['low','medium','high'], ARRAY[40000,40000,40000]), COALESCE( ELEMENT_AT( other_map, name ), 'low' ) ) ) AS v1, " +
+                "FILTER( MAP_VALUES(map1), x -> x >= ELEMENT_AT( MAP(ARRAY['low','medium','high'], ARRAY[40000,40000,40000]), COALESCE( ELEMENT_AT( other_map, name ), 'low' ) ) ) AS v2 " +
+                "FROM (select MAP(ARRAY['low'], ARRAY[100000]) map1, '' name) CROSS JOIN ( SELECT MAP() AS other_map ) risk_map";
+        MaterializedResult result = computeActual(sql);
```

Comment:
you should be able to call AbstractTestQueryFramework.assertQuery(sql), and it will run in presto and H2 and compare the results.  Or if it's not supported in H2 you can do assertQuery(testQuery, expectedResultsQuery), and for the expected results Query do a simple SELECT with no from or SELECT FROM values

Reply:
Done, changed the check code to `assertQuery(sql, "VALUES (100000, 100000)");`